src/client/components/App.tsx

```diff
@@ -0,0 +1,96 @@
+import React, {
+  useState,
+  useEffect,
+  useReducer,
+  ReactElement,
+  SFC,
+} from 'react';
+import { hot } from 'react-hot-loader/root';
+import { User } from 'server/models';
+import { AppMessage } from 'client/classes';
+import {
+  MessageContext,
+  messageReducer,
+  UserContext,
+} from 'client/context';
+import { getCurrentUser } from 'client/api';
+import { Message } from './layout';
+
+/**
+ * The primary app component. Fetches the current user from the server when it
+ * mounts, then saves it to the UserContext to pass down to other components
+ */
+
+const ColdApp: SFC = (): ReactElement => {
+  /**
+   * Hook for maintaining the currently selected user
+   * */
+
+  const [currentUser, setUser] = useState();
+
+  /**
+   * Set up the local reducer for maintaining the current app-wide message
+   * queue. The dispatchMessage function will be passed down through the
+   * Message Context Provider
+   * */
+
+  const [{ currentMessage, queue }, dispatchMessage] = useReducer(
+    messageReducer,
+    {
+      queue: [],
+      currentMessage: undefined,
+    }
+  );
+
+  /**
+   * Get the currently authenticated user from the server on launch.
+   * If it fails, display a message for the user
+   */
+
+  useEffect((): void => {
```

Comment:
Does it make sense to use `async`/`await` here instead?

Reply:
On the surface it does make sense, but React throws an error if you return a promise from the callback to `useEffect()` so you basically have to wrap your `async` function in a synchronous function and the whole things gets even messier. I don't love the `then(..).then(...).catch(...)` syntax but I feel like it's at least easier to read at a glance.

See [this issue](https://github.com/facebook/react/issues/14326) for more details.